bluesky/run_engine.py

```diff
@@ -1489,9 +1494,10 @@ def _kickoff(self, msg):
         ret = obj.kickoff(*msg.args, **kwargs)
 
         p_event = asyncio.Event(loop=self.loop)
+        pardon_failures = self._pardon_failures
 
         def done_callback():
-            if not ret.success:
+            if not ret.success and not pardon_failures.is_set():
```

Comment:
I think this needs to be passed into `call_soon_threadsafe` and this check done in `RunEngine._failed_status` as this callback is run in the pyepics thread.  I think we might want to refactor this to push all of the logic of this callback into a `RunEngine._status_object_completed` callback so that the `p_event.set` call can be run in the event loop.

Tracking all of the status-related tasks that end up on the loop is probably a good idea, right now we are losing track of the tasks to set the `p_event`.